src/test/java/com/google/devtools/build/lib/analysis/RunfilesRepoMappingManifestTest.java

```diff
@@ -321,4 +324,87 @@ public void actionRerunsOnRepoMappingChange_newEntry() throws Exception {
+    scratch.overwriteFile(
+        moduleRoot.getRelative("aaa~1.0/BUILD").getPathString(),
+        "load('@bare_rule//:defs.bzl', 'bare_binary')",
+        "bare_binary(name='aaa',data=['@symlinks'])");
+
+    registry.addModule(
+        createModuleKey("symlinks", "1.0"),
+        "module(name='symlinks',version='1.0')",
+        "bazel_dep(name='ddd',version='1.0')");
+    scratch.overwriteFile(moduleRoot.getRelative("symlinks~1.0/WORKSPACE").getPathString());
+    scratch.overwriteFile(
+        moduleRoot.getRelative("symlinks~1.0/defs.bzl").getPathString(),
+        "def _symlinks_impl(ctx):",
+        "  runfiles = ctx.runfiles(",
+        "    symlinks = {'path/to/pkg/symlink': ctx.file.data},",
+        "    root_symlinks = {ctx.label.workspace_name + '/path/to/pkg/root_symlink': ctx.file.data},",
+        "  )",
+        "  return DefaultInfo(runfiles=runfiles)",
+        "symlinks = rule(",
+        "  implementation=_symlinks_impl,",
+        "  attrs={'data':attr.label(allow_single_file=True)},",
+        ")");
+    scratch.overwriteFile(
+        moduleRoot.getRelative("symlinks~1.0/BUILD").getPathString(),
+        "load('//:defs.bzl', 'symlinks')",
+        "symlinks(name='symlinks',data='@ddd')");
+
+    registry.addModule(
+        createModuleKey("ddd", "1.0"),
+        "module(name='ddd',version='1.0')",
+        "bazel_dep(name='bare_rule',version='1.0')");
+    scratch.overwriteFile(moduleRoot.getRelative("ddd~1.0/WORKSPACE").getPathString());
+    scratch.overwriteFile(
+        moduleRoot.getRelative("ddd~1.0/BUILD").getPathString(),
+        "load('@bare_rule//:defs.bzl', 'bare_binary')",
+        "bare_binary(name='ddd')");
+
+    RunfilesSupport runfilesSupport = getRunfilesSupport("@aaa~1.0//:aaa");
+    List<String> runfilesPaths =
+        runfilesSupport
+            .getRunfiles()
+            .getRunfilesInputs(reporter, Location.BUILTIN, runfilesSupport.getRepoMappingManifest())
+            .keySet()
+            .stream()
+            .map(PathFragment::getPathString)
+            .collect(toImmutableList());
+    assertThat(runfilesPaths)
+        .containsExactly(
+            "aaa~1.0/aaa",
+            "_main/external/aaa~1.0/aaa",
+            "_main/path/to/pkg/symlink",
+            "symlinks~1.0/path/to/pkg/root_symlink",
+            "_repo_mapping");
+
+    assertThat(getRepoMappingManifestForTarget("@aaa~1.0//:aaa"))
+        .containsExactly(
+            // @aaa~1.0 contributes the top-level executable to runfiles.
+            "aaa~1.0,aaa,aaa~1.0",
+            // The symlink is staged under the main repository's runfiles directory and aaa has a
+            // repo mapping entry
+            // for it.
+            "aaa~1.0,my_module,_main",
+            // @symlinks~1.0 appears as the first segment of a root symlink.
+            "aaa~1.0,symlinks,symlinks~1.0",
+            "symlinks~1.0,symlinks,symlinks~1.0")
+        .inOrder();
```

Comment:
More just a question -- the order matters in the repo mapping?

Reply:
It doesn't matter, but is specified as being lexicographic. I can remove this assertion if you think it just adds noise

Reply:
Because it's testing an attribute that is specified, I'm fine with either.